File: EasyObjects.Tests/Tests/SqlServer/SqlXmlFixture.vb
Imports System
Imports NUnit.Framework
Imports NCI.EasyObjects

NameSpace EasyObjects.Tests.SQL

	<TestFixture()> _
	Public Class SqlXmlFixture
		
		Dim aggTest As AggregateTest = New AggregateTest
		Dim aggClone As AggregateTest = New AggregateTest

		<TestFixtureSetUp()> _
		Public Sub Init()
            aggTest.DatabaseInstanceName = "SQLUnitTests"
            aggTest.DefaultCommandType = Data.CommandType.Text
            UnitTestBase.RefreshDatabase()
        End Sub

        <SetUp()> _
        Public Sub Init2()
            aggTest.FlushData()
        End Sub

        <Test()> _
        Public Sub SerializeDeserialize()
            aggTest.LoadAll()
            aggTest.LastName = "Griffinski"
            aggTest.GetChanges()
            Dim str As String = aggTest.Serialize
            aggClone.Deserialize(str)
            Assert.AreEqual(1, aggClone.RowCount)
            Assert.AreEqual("Modified", aggClone.RowState.ToString)
            Assert.AreEqual("Griffinski", aggClone.s_LastName)
        End Sub

        <Test()> _
        Public Sub ToXmlFromXml()
            aggTest.LoadAll()
            aggTest.LastName = "Griffinski"
            aggTest.GetChanges()
            Dim str As String = aggTest.ToXml
            aggClone.FromXml(str)
            Assert.AreEqual(1, aggClone.RowCount)
            Assert.AreEqual("Added", aggClone.RowState.ToString)
            Assert.AreEqual("Griffinski", aggClone.s_LastName)
        End Sub
    End Class
End Namespace

File: EasyObjects.Tests/CS/Tests/SqlServer/SqlXmlFixture.cs
using System;
using NUnit.Framework;
using NCI.EasyObjects;

namespace NCI.EasyObjects.Tests.SQL
{
	[TestFixture]
	public class SqlXmlFixture
	{
		AggregateTest aggTest = new AggregateTest();
		AggregateTest aggClone = new AggregateTest();
		
		[TestFixtureSetUp]
		public void Init()
		{
			aggTest.DatabaseInstanceName = "SQLUnitTests";
			aggTest.DefaultCommandType = System.Data.CommandType.Text;
			UnitTestBase.RefreshDatabase();
		}

		[SetUp]
		public void Init2()
		{
			aggTest.FlushData();
		}

		[Test]
		public void SerializeDeserialize()
		{
			aggTest.LoadAll();
			aggTest.LastName = "Griffinski";
			aggTest.GetChanges();
			string str = aggTest.Serialize();
			
			aggClone.Deserialize(str);
			Assert.AreEqual(1, aggClone.RowCount);
			Assert.AreEqual("Modified", aggClone.RowState().ToString());
			Assert.AreEqual("Griffinski", aggClone.s_LastName);
		}

		[Test]
		public void ToXmlFromXml()
		{
			aggTest.LoadAll();
			aggTest.LastName = "Griffinski";
			aggTest.GetChanges();
			string str = aggTest.ToXml();
			
			aggClone.FromXml(str);
			Assert.AreEqual(1, aggClone.RowCount);
			Assert.AreEqual("Added", aggClone.RowState().ToString());
			Assert.AreEqual("Griffinski", aggClone.s_LastName);
		}

	}
}
